packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -65,8 +65,8 @@ abstract class MacOSApp extends ApplicationPackage {
       printError('Invalid prebuilt macOS app. Does not contain Info.plist.');
       return null;
     }
-    final String id = plist.getValueFromFile(plistPath, plist.kCFBundleIdentifierKey);
-    final String executableName = plist.getValueFromFile(plistPath, plist.kCFBundleExecutable);
+    final String id = const PlistUtils().getValueFromFile(plistPath, PlistUtils.kCFBundleIdentifierKey);
+    final String executableName = const PlistUtils().getValueFromFile(plistPath, PlistUtils.kCFBundleExecutable);
```

Comment:
This is parsing the entire file twice in a row; it should use `parseFile` instead.

Alternately, maybe instead of being a util-holding faux class, `PlistFile` should be a class that's initialized from a filename, and then can be queried, to make code like this simpler?

Reply:
I changed it to use `parseFile()` (good call - this was just a raw port from the old calls).

The reason for the faux-class is to allow it to be injected into the application context and mocked in tests.  If we changed it to be initialized to a filename, then we'd need another level of indirection whereby callers injected a PlistParserFactor - doesn't seem worth it.